### ISSUE TITLE
Can't manage to install this plugin

### ISSUE THREAD

**Issue by drug007** [Groterik/qtcreator-dlangeditor#14]
What shoul I do to install it?

**Comment by Groterik** [comment]
In a nutshell:

1. You should have the latest stable QtCreator installed and its sources.
1. Clone this repository and compile the plugin.
1. Download or build DCD.
1. Open QtCreator and enable the plugin.
1. Open QtCreator's settings and configure the plugin.

If you have any questions, issues or suggestions don't hesitate to ask me.

**Comment by drug007** [comment]
I have built plugin itself, but can't understand how to install it. Also I have built example of simple qt creator plugin (using these [instructions](http://habrahabr.ru/post/135289/)) and this plugin was installed automatically - it appeared in menu. So first question is how I can understand that plugin has been installed and how can I install it if it haven't. 
Instructions for other your [project](https://github.com/Groterik/qtcreator-dubmanager#installation) say that I have to copy two files in specific directory, but after successful building I have only one of these files - .so lib.
I  use qt creator 3.5.81 (IIRC - I did it at home, now I'm at office), but I isn't installed, although I use path to these version, system wide version is another, could it be the reason?

**Comment by Groterik** [comment]
Actually if you built it using default configuration, the plugin would be placed into default directory that QtCreator uses to load user's plugins. But it depends on the environment. The output directory is shown during `qmake` step. Also it is possible to set plugin directory manually with `-pluginpath <path_to_dir_with_so>` command line argument for QtCreator according to [the doc page](http://doc.qt.io/qtcreator/creator-cli.html). I believe `.so` is enough for recent QtC. As the first step I would suggest to use this argument.

**Comment by drug007** [comment]
Thanks for help, I'll try it at home. But what changes after plugin installation? For example, ability to create D project appears or something else.

**Comment by Groterik** [comment]
If the plugins appear in QtC `Help -> About plugins` and can be enabled, then special settings will be accessible in the menu `Tools -> Options -> D` and `Build & Run -> DubManager`. You will probably need  to restart QtC. It will be a good sign that everything is correct and is ready to use.
DubManager provides a project tree, build and run configurations, errors and warnings for DUB projects, as well as ability to create DUB project via QtC's wizard.
DlangEditor provides an autocompletion, function tooltips, help and so on.

**Comment by drug007** [comment]
DlangEditor Plugin installs, I have access to settings in the menu `Tools->Options->D`, but DubManager doesn't compile. Can I create D project without DubManager plugin?

**Comment by stonemaster** [comment]
You can always create a DUB project on command line using `dub init test` and then opening the D files manually in Qt Creator.

**Comment by kindlychung** [comment]
The documentation seems a bit vague, could you provide a more concrete setup example? Thanks for your hard work. 

For example, what is wrong with the following modification in dlangeditor.pro?

```
## set the QTC_SOURCE environment variable to override the setting here
QTC_SOURCE = /Users/kaiyin/Qt/qt-creator
QTCREATOR_SOURCES = $$QTC_SOURCE
isEmpty(QTCREATOR_SOURCES):QTCREATOR_SOURCES=/usr/src/qtcreator
!exists($$QTCREATOR_SOURCES):\
    error("Set variable QTC_SOURCE to the QtCreator's sources path (current path is \"$$QTC_SOURCE\")")
```

Now I understand QTC_SOURCE is the soure of qt creator (https://github.com/qtproject/qt-creator), but what does QTC_BUILD mean?

Thanks again.

**Comment by Groterik** [comment]
If you are going to build the plugin from sources you should set QTC_SOURCE and QTC_BUILD to the directories with QtCreator sources and binaries respectively. Source directory should contain qtcreator.pro file and binary directory contains ```bin``` subdirectory with ```Qt Creator.app``` under Mac OS X. Build directory is usually installed with Qt framework.
I will try to reanimate Mac OS X building/publishing to avoid such a nontrivial and time-consuming step.

**Comment by kindlychung** [comment]
Ok, I see. QTC_BUILD should be "~/Qt/5.5/clang_64" in my case. After using 3.6 source, I still got some errors:

```
13:41:06: Running steps for project dlangeditor...
13:41:06: Configuration unchanged, skipping qmake step.
13:41:06: Starting: "/usr/bin/make" 
/Users/kaiyin/Qt/5.5/clang_64/bin/qmake -spec macx-clang CONFIG+=debug CONFIG+=x86_64 CONFIG+=qml_debug -o Makefile dlangeditor.pro
Cannot read /Users/kaiyin/Qt/qt-build/src/qtcreatorplugin.pri: No such file or directory
Project MESSAGE: Your QtCreator's sources version is 
Project MESSAGE: Your QtCreator's sources major version is 
Project MESSAGE: Your QtCreator's sources minor version is 
Project MESSAGE: Plugin output path is 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -c -pipe -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -mmacosx-version-min=10.7 -Wall -W -fPIC -DQTCREATOR_MAJOR_VERSION= -DQTCREATOR_MINOR_VERSION= -DQT_QML_DEBUG -DQT_GUI_LIB -DQT_CORE_LIB -I. -Isrc -Ithirdparty/msgpack/include -I/Users/kaiyin/Qt/5.5/clang_64/lib/QtGui.framework/Headers -I/Users/kaiyin/Qt/5.5/clang_64/lib/QtCore.framework/Headers -I. -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/OpenGL.framework/Headers -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/AGL.framework/Headers -I/Users/kaiyin/Qt/5.5/clang_64/mkspecs/macx-clang -F/Users/kaiyin/Qt/5.5/clang_64/lib -o dlangeditorplugin.o src/dlangeditorplugin.cpp
In file included from src/dlangeditorplugin.cpp:1:
src/dlangeditorplugin.h:6:10: fatal error: 'extensionsystem/iplugin.h' file not found
#include <extensionsystem/iplugin.h>
         ^
1 error generated.
make: *** [dlangeditorplugin.o] Error 1
13:41:07: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project dlangeditor (kit: Desktop Qt 5.5.1 clang 64bit)
When executing step "Make"
13:41:07: Elapsed time: 00:01.
```

**Comment by Groterik** [comment]
Something is wrong with your QTC_SOURCE dir. Please check if it is set to QtCreator's source directory containing qtcreator.pro file.

**Comment by kindlychung** [comment]
Ok. Correct the qt source dir. This time got a different error: 

```
In file included from src/codemodel/dastedmodel.cpp:1:
src/codemodel/dastedmodel.h:52:10: warning: 'appendIncludePaths' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    void appendIncludePaths(const QStringList &includePaths);
         ^
src/codemodel/dmodel.h:102:18: note: overridden virtual function is here
    virtual void appendIncludePaths(const QStringList &includePaths) = 0;
                 ^
In file included from src/codemodel/dastedmodel.cpp:2:
src/codemodel/dastedmessages.h:7:10: fatal error: 'msgpack.hpp' file not found
#include <msgpack.hpp>
         ^
1 warning and 1 error generated.
make: *** [.obj/debug-shared/dastedmodel.o] Error 1
22:05:29: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project dlangeditor (kit: Desktop Qt 5.5.1 clang 64bit)
When executing step "Make"
22:05:29: Elapsed time: 00:16.
```

**Comment by Groterik** [comment]
The plugins are building now under Mac OSX on travis. The binaries can be found at [Bintray](https://bintray.com/groterik/qtcreator-dlang-tools/). Please ping me if you have any issues.
If you are curious what the error was: you should have done `git submodule update --init` to fetch all the external modules to your local repository that are used by the project (currently, this is msgpack only).

**Comment by bioglaze** [comment]
The installation directory mentioned in the Readme for OS X didn't exist on my machine (Qt Creator 4.0.0) so I had to copy the plugin into /Applications/Qt Creator.app/Contents/PlugIns/libDlangEditor.dylib. Also, the binary distribution didn't contain a pluginspec file.

**Comment by Groterik** [comment]
Thanks, I'll check it soon.

**Comment by Groterik** [comment]
Correct, this directory mentioned in Readme isn't created with QtCreator installation but this is a default per-user directory for plugins though. So, you could create it and copy the plugin there (`/Users/$USER/Library/Application\ Support/QtProject/Qt\ Creator/plugins/4.0.1/` in my case).
You can always use `-pluginpath` flag for QtCreator binary to specify custom plugin directory.
I'm not sure if the pluginspec is needed for anything, my understanding was that it is compiled into the plugin binary.
BTW, dubmanager plugin building and uploading has been just fixed for OSX.
